source/reference/configure-shell-settings.txt

```diff
@@ -91,21 +91,96 @@ Supported ``property`` parameters
      - boolean
      - false
      - Controls display of a stack trace along with error messages.
-     
+
+.. _customize-the-mongosh-prompt:
+
+Customize the :binary:`~bin.mongosh` Prompt
+-------------------------------------------
+
+The ``prompt`` variable can store strings and JavaScript code to
+customize the :binary:`~bin.mongosh` prompt. Use a function that
+returns a string to display dynamic information in the prompt.
+
+Custom prompts are not stored when you exit :binary:`~bin.mongosh`. Add
+the code for your custom prompt to :ref:`.mongoshrc.js <mongoshrc-js>`
+to set the prompt each time you start up the mongo shell.
+
+Display Line Numbers
+~~~~~~~~~~~~~~~~~~~~
+
+This code will dynamically update the :binary:`~bin.mongosh` prompt
+to display line numbers:
+
+.. code-block:: javascript
+
+   cmdCount = 1;
+   prompt = function() {
+                return (cmdCount++) + "> ";
+            }
+
+The prompt will look like this:
+
+.. code-block:: javascript
+
+   1> show collections
+   2> use test
+   3>
+
+Display Database and Hostname
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use a function like this one to display the database and hostname in
+the :binary:`~bin.mongosh` prompt:
+
+.. code-block:: javascript
+
+   host = db.serverStatus().host;
+   prompt = function() {
+                return db.getName()+"@"+host+"> ";
```

Comment:
```suggestion
                return `${db.getName()}@${host}> `;
```

(also just a best practice thing)

It might be worth calling out here that this will update the database name, but not the hostname, if `db` is reassigned (e.g. via `db = connect(...)`).

A version that accounts for that by caching the hostname might look like this:

```js
{
  const hostnameSymbol = Symbol('hostname');
  prompt = () => {
    if (!db[hostnameSymbol]) 
      db[hostnameSymbol] = db.serverStatus().host;
    return `${db.getName()}@${db[hostnameSymbol]}> `;
  };
}
```

I think it’s worth writing it out like this – including the block-scoping and the usage of `const` – so that we encourage users to write idiomatic and reliable code, even if it’s just for prompt customization.

Reply:
Thanks Anna, that's very helpful